R/mod_export_count_table.R

```diff
@@ -0,0 +1,208 @@
+          checkmate::check_list(dataset),
+          checkmate::check_subset(names(dataset), choices = c("df", "meta")),
+          checkmate::check_data_frame(dataset[["df"]]),
+          all(c("n_denominator", "hierarchy", "hier_lvl_col") %in% names(dataset[["meta"]])),
+          combine = "and"
+        )
+
+        # If there is no dataset being displayed
+        if (is.null(dataset)) {
+          shinyFeedback::showToast(
+            "info", "There is no dataset displayed currently. This may arise due to your filter choices or
+            because the dataset is still loading.",
+            .options = list(positionClass = "toast-top-right")
+          )
+        } else {
+          shiny::showModal(
+            shiny::modalDialog(
+              shiny::tagList(
+                shiny::textInput(
+                  ns(EXP$ID$FILENAME_BOX),
+                  "Insert file name",
+                  value = "file_name"
+                ),
+                shiny::checkboxInput(
+                  inputId = ns(EXP$ID$DATAPROTECT_BOX),
+                  label = shiny::uiOutput(ns("label_id"), inline = TRUE),
+                  width = "100%"
+                )
+              ),
+              title = "Download box",
+              footer = list(shiny::fluidRow(
+                shiny::column(5, shinyjs::disabled(shiny::downloadButton(ns(EXP$ID$DOWNLOAD_BUTTON))), offset = 5),
+                shiny::column(1, shiny::modalButton("Cancel"))
+              )),
+              easyClose = TRUE
+            )
+          )
+        }
+      })
+
+
+      # Add/remove checkbox warning
+      checkbox_label <- shiny::eventReactive(input[[EXP$ID$DATAPROTECT_BOX]], {
+        if (input[[EXP$ID$DATAPROTECT_BOX]]) {
+          return(paste(EXP$MSG$DATAPROTECT_LABEL, intended_use_label))
+        } else {
+          return(shiny::tags$embed(
+            paste(EXP$MSG$DATAPROTECT_LABEL, intended_use_label),
+            shiny::br(shiny::tags$span(
+              style = paste("color:", "#CC3333"),
+              EXP$MSG$DATAPROTECT_WARN
+            ))
+          ))
+        }
+      })
+
+      # Update checkbox label separately (to allow testing of logic)
+      output$label_id <- shiny::renderUI(checkbox_label())
+
+
+      # Check if we should enable/disable download button
+      download_enable <- shiny::eventReactive(
+        c(input[[EXP$ID$FILENAME_BOX]], input[[EXP$ID$DATAPROTECT_BOX]]),
+        {
+          if (
+            (input[[EXP$ID$FILENAME_BOX]] != "") &
+              input[[EXP$ID$DATAPROTECT_BOX]]
+          ) { # nolint end
+            return(TRUE)
+          } else {
+            return(FALSE)
+          }
+        }
+      )
+
+      # Enable/disable download button separately (to allow testing of logic)
+      shiny::observeEvent(download_enable(), {
+        if (download_enable()) {
+          shinyjs::enable(EXP$ID$DOWNLOAD_BUTTON)
+        } else {
+          shinyjs::disable(EXP$ID$DOWNLOAD_BUTTON)
+        }
+      })
```

Comment:
Can we compact these lines in a single observe? `download_enable` is not used anywhere else as far as I can see